src/graph/expression_operators.h

```diff
@@ -3,75 +3,346 @@
+
+ * @see PReLUNodeOp
+ */
 Expr leakyrelu(Expr a);
-Expr leakyrelu(const std::vector<Expr>&);
 
+/**
+ * @copybrief leakyrelu
+ * @warning not implemented
+ */
+Expr leakyrelu(const std::vector<Expr>& nodes);
+
+/**
+ * @brief Parametric Rectified Linear Unit (PReLU)
+ *
+ * Computes the <a href="https://en.wikipedia.org/wiki/Rectifier_(neural_networks)#Parametric_ReLU">
+ * Parametric ReLU</a> activation for the expression
+ * \f[
+ *   \operatorname{leakyrelu}(x) =
+ *   \begin{cases}
+ *     \alpha x & \text{if } x \leq 0 \\
+ *     x & \text{if } x > 0
+ *   \end{cases}
+ * \f]
+
+ * @see PReLUNodeOp
+ * @note @p alpha is **not** trainable.
+ */
 Expr prelu(Expr a, float alpha = 0.01);
-Expr prelu(const std::vector<Expr>&, float alpha = 0.01);
 
+/**
+ * @copybrief prelu
+ * @warning not implemented
+ */
+Expr prelu(const std::vector<Expr>&, float alpha = 0.01);
+/** @} */
+
+/**
+ * @addtogroup graph_ops_mathematical Mathematical
+ * @ingroup graph_ops
+ * @brief Performs mathematical operations in the expression graph
+ *
+ * @{
+*/
+/**
+ * @name Exponentiation and Logarithmic functions
+ * @{
+ */
+/**
+* @brief Natural logarithm
+*
+* Computes the element-wise natural logarithm of the expression: \f$ \log(a) \f$
+* @see LogNodeOp
+*/
 Expr log(Expr a);
-Expr exp(Expr a);
 
+/**
+ * @brief Natural exponentiation
+ *
+ * Computes the element-wise natural logarithm of the expression: \f$ e^a \f$
+ * @see ExpNodeOp
+ */
+Expr exp(Expr a);
+/** @} */
+
+/**
+ * @name Trigonometric functions
+ * @{
+ */
+/**
+* @brief Sine
+*
+* Computes the element-wise sine of the expression: \f$ \sin(a) \f$
+* @see SinNodeOp
+*/
 Expr sin(Expr a);
-Expr cos(Expr a);
-Expr tan(Expr a);
 
-Expr clip(Expr a, float c);
+/**
+* @brief Cosine
+*
+* Computes the element-wise cosine of the expression: \f$ \cos(a) \f$
+* @see CosNodeOp
+*/
+Expr cos(Expr a);
 
+/**
+* @brief Tangent
+*
+* Computes the element-wise tangent of the expression: \f$ \tan(a) \f$
+* @see TanNodeOp
+*/
+Expr tan(Expr a);
+/** @} */
+/** @} */
+
+
+/**
+ * @addtogroup graph_ops_arithmetic Arithmetic
+ * @ingroup graph_ops
+ * @brief Performs arithmetic in the expression graph
+ *
+ * @{
+*/
+
+/**
+ * Returns \f$ -a \f$
+ * @see NegNodeOp for implementation.
+ */
+///@{
 Expr operator-(Expr a);
+///@}
 
 /*********************************************************/
 
-Expr operator+(Expr a, Expr b);
-Expr operator+(float a, Expr b);
-Expr operator+(Expr a, float b);
-
-Expr operator-(Expr a, Expr b);
-Expr operator-(float a, Expr b);
-Expr operator-(Expr a, float b);
-
-Expr operator*(Expr a, Expr b);
-Expr operator*(float a, Expr b);
-Expr operator*(Expr a, float b);
+/**
+ * @name Addition
+ * @brief Performs \f$ a + b \f$ in the expression graph
+ *
+ * @{
+*/
+Expr operator+(Expr a, Expr b);   //!< @see Implementation in PlusNodeOp
+Expr operator+(float a, Expr b);  //!< @see Implementation in ScalarAddNodeOp
+Expr operator+(Expr a, float b);  //!< @see Implementation in ScalarAddNodeOp
```

Comment:
#788 consistently uses `///<` for in-line documentation. I think we should choose one, and my suggestion is `///<` as it's less distinct characters.

Reply:
Done. To that end, once we've coalesced on a consistent style, we should document it as a reference.